### ISSUE TITLE
Exception with optional tracked property

### ISSUE THREAD

**Issue by nwhittaker** [kltan/ember-tracked-nested#2]
The other issue I ran into was an exception when the tracked property is optional and initially `undefined`:

```
TypeError: Cannot read property 'call' of null
    at TestComponent.desc.initializer (http://localhost:7357/assets/vendor.js:71830:29)
    at getter (http://localhost:7357/assets/vendor.js:58964:29)
    at TestComponent.get [as obj] (http://localhost:7357/assets/vendor.js:16419:19)
    at TestComponent.get value [as value] (http://localhost:7357/assets/tests.js:288:36)
    at getPossibleMandatoryProxyValue (http://localhost:7357/assets/vendor.js:13811:19)
    at _getProp (http://localhost:7357/assets/vendor.js:13876:17)
    at http://localhost:7357/assets/vendor.js:49296:45
    at http://localhost:7357/assets/vendor.js:49244:37
    at track (http://localhost:7357/assets/vendor.js:58933:7)
    at valueForRef (http://localhost:7357/assets/vendor.js:49243:44)
```

Added a failing test at https://github.com/nwhittaker/ember-tracked-nested/commit/ff3b7de0088eeb9a837f07b9abd85b6b32a38829. Also happy to submit a PR with a possible fix, if desired.

**Comment by kltan** [comment]
Thanks for catching it! Will take a stab at this issue.

**Comment by kltan** [comment]
1.0.4 should resolve the issue. Please reopen if not fixed.

**Comment by nwhittaker** [comment]
Thanks for addressing this and #1 so quickly! I have a follow-on question for this issue:

```js
@trackedNested obj;

update() {
  this.obj = { foo: { bar: 1 } };
}
```

Given the above snippet, if I run `update()`, changes to `obj.foo.bar` are no longer tracked.

I see that I can restore the tracking behavior with this change:

```diff
update() {
-  this.obj = { foo: { bar: 1 } };
+  this.obj = nested({ foo: { bar: 1 } }, this, 'obj').data;
}
```

Is using `nested().data` the intended API for this scenario?

**Comment by kltan** [comment]
Thanks. That is also fixed as of 1.0.11
